### PR TITLE
[MODFEE-21] Send patron notice for cancellation of fee/fine charge (due to error)

### PR DESCRIPTION
**Pull request by oleksandr-dekin**
**Purpose**
Enable sending of patron notices for error.

**Approach**
Extend the list of fee/fine actions with error.
Add token "fee.reasonForCancellation" to template

**Links**
[MODFEE-21](https://issues.folio.org/browse/MODFEE-21)

### CODE REVIEW

File: src/test/java/org/folio/rest/domain/FeeFineNoticeContextTest.java [hunk 1]
```diff
@@ -66,6 +66,7 @@ public FeeFineNoticeContextTest(Owner owner, Feefine feeFine, Feefineaction feeF
     Feefineaction waivedPartially = createActionWithType("Waived partially");
     Feefineaction transferredFully = createActionWithType("Transferred fully");
     Feefineaction transferredPartially = createActionWithType("Transferred partially");
+    Feefineaction canceledAsError = createActionWithType("Cancelled as error");
```
Comment (OleksandrVidinieiev) [inline src/test/java/org/folio/rest/domain/FeeFineNoticeContextTest.java]:
```suggestion
    Feefineaction cancelledAsError = createActionWithType("Cancelled as error");
```

File: src/test/java/org/folio/rest/impl/FeeFineActionsAPITest.java [hunk 2]
```diff
@@ -247,7 +247,8 @@ private JsonObject createNoticeContext(String feeFineType,
         .put("actionAmount", amountAction)
         .put("actionDateTime", dateAction)
         .put("balance", balance)
-        .put("actionAdditionalInfo", ""));
+        .put("actionAdditionalInfo", "")
+        .put("reasonForCancellation","for staff"));
```
Comment (OleksandrVidinieiev) [inline src/test/java/org/folio/rest/impl/FeeFineActionsAPITest.java]:
```suggestion
        .put("reasonForCancellation", "for staff"));
```

File: src/test/java/org/folio/rest/domain/FeeFineNoticeContextTest.java (hunk 1 above)
Comment (OleksandrVidinieiev) [reply src/test/java/org/folio/rest/domain/FeeFineNoticeContextTest.java]: This new object does not seem to be used anywhere.

File: src/test/java/org/folio/rest/impl/FeeFineActionsAPITest.java [hunk 3]
```diff
@@ -247,7 +247,8 @@ private JsonObject createNoticeContext(String feeFineType,
         .put("actionAmount", amountAction)
         .put("actionDateTime", dateAction)
         .put("balance", balance)
-        .put("actionAdditionalInfo", ""));
+        .put("actionAdditionalInfo", "")
```
Comment (OleksandrVidinieiev) [inline src/test/java/org/folio/rest/impl/FeeFineActionsAPITest.java]: Why not add a non-empty value to `actionAdditionalInfo`?

File: src/test/java/org/folio/rest/domain/FeeFineNoticeContextTest.java (hunk 1 above)
Comment (oleksandr-dekin) [reply src/test/java/org/folio/rest/domain/FeeFineNoticeContextTest.java]: Fixed

File: src/test/java/org/folio/rest/impl/FeeFineActionsAPITest.java (hunk 2 above)
Comment (oleksandr-dekin) [reply src/test/java/org/folio/rest/impl/FeeFineActionsAPITest.java]: Changed

File: src/test/java/org/folio/rest/impl/FeeFineActionsAPITest.java (hunk 3 above)
Comment (oleksandr-dekin) [reply src/test/java/org/folio/rest/impl/FeeFineActionsAPITest.java]: Added coment

File: src/test/java/org/folio/rest/domain/FeeFineNoticeContextTest.java (hunk 1 above)
Comment (OleksandrVidinieiev) [reply src/test/java/org/folio/rest/domain/FeeFineNoticeContextTest.java]:
```suggestion
    Feefineaction cancelledAsError = createActionWithType("Cancelled as error");
```

**Pull request merged**